server/filestore.go

```diff
@@ -3109,6 +3103,57 @@ func (fs *fileStore) dmapEntries() int {
 	return total
 }
 
+// Fixed helper for iterating.
+func subjectsEqual(a, b string) bool {
+	return a == b
+}
+
+func subjectsAll(a, b string) bool {
+	return true
+}
+
+func compareFn(subject string) func(string, string) bool {
+	if subject == _EMPTY_ || subject == ">" {
+		return subjectsAll
+	}
+	if subjectHasWildcard(subject) {
+		return subjectIsSubsetMatch
+	}
+	return subjectsEqual
+}
+
+// PurgeEx will remove messages based on subject filters, sequence and number of messages to keep.
+// Will return the number of purged messages.
+func (fs *fileStore) PurgeEx(subject string, sequence, keep uint64) (purged uint64, err error) {
+	if subject == _EMPTY_ || subject == ">" && keep == 0 {
```

Comment:
Could also check if the subject is `*` and it the ingest subject is single token

Reply:
I thought that as well at first but I don't know the true makeup of the diversity of subjects in the stream since they could be delivered via source or mirror etc.

Reply:
Oh yeah good point didn’t think about that